validators.py

```diff
@@ -0,0 +1,19 @@
+from functools import wraps
+
+import sqlalchemy
+from fastapi import HTTPException, status
+
+
+def unique_constraint_handler(func):
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        try:
+            return func(*args, **kwargs)
+        except sqlalchemy.exc.IntegrityError:
+            raise HTTPException(
+                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                detail="Unique constraint violation. "
+                "The resource already exists.",
+            )
+
+    return wrapper
```

Comment:
I wasn't sure about this decision. 
Is it a bad idea to use decorators to catch errors because if there are more errors, then there will be too many decorators on top of functions?)) Is it better to use separate functions and use them inside endpoints?

Reply:
I wasn't sure about this decision...